src/lib/context.ts

```diff
@@ -1,9 +1,34 @@
 import { createContext } from "react";
+import { Plurals } from "../components/Ditto";
+
+interface StructuredTextData {
+  plurals: Plurals;
+  text: string;
+  variables: Variables;
+}
+type FlatTextData  = string
+
+export interface TextData  {
+  plurals: Plurals;
+  text: string;
+  variables: VariableData;
```

Comment:
Should this be something like this?
```ts
variables: {
  [variableName: string]: VariableData;
}

Reply:
no the variableData is what we get from ditto, while Variables is what is passed in my the user input